examples/digital_fingerprinting/production/morpheus/dfp_modules_pipeline.py

```diff
@@ -0,0 +1,187 @@
+          "See fsspec documentation for list of possible options."),
+)
+@click.option('--tracking_uri',
+              type=str,
+              default="http://mlflow:5000",
+              help=("The MLflow tracking URI to connect to the tracking backend."))
+def run_pipeline(log_type: str,
+                 workload_type: str,
+                 train_users: str,
+                 skip_user: typing.Tuple[str],
+                 only_user: typing.Tuple[str],
+                 start_time: datetime,
+                 duration: str,
+                 cache_dir: str,
+                 log_level: int,
+                 sample_rate_s: int,
+                 tracking_uri,
+                 **kwargs):
+
+    derive_args = DeriveArgs(skip_user,
+                             only_user,
+                             start_time,
+                             log_level,
+                             cache_dir,
+                             sample_rate_s,
+                             duration,
+                             log_type,
+                             tracking_uri,
+                             workload_type,
+                             train_users)
+
+    derive_args.init()
+
+    userid_column_name = "username"
+    timestamp_column_name = "timestamp"
+
+    config: Config = generate_ae_config(log_type, userid_column_name, timestamp_column_name)
+
+    schema_builder = SchemaBuilder(config, log_type)
+    schema: Schema = schema_builder.build_schema()
+
+    config_generator = ConfigGenerator(config, derive_args, schema)
+
+    module_config = config_generator.get_module_config()
+
+    workload = module_config.get("workload")
+    output_port_count = module_config.get("output_port_count")
+
+    # Create a pipeline object
+    pipeline = Pipeline(config)
+
+    source_stage = pipeline.add_stage(MultiFileSource(config, filenames=list(kwargs["input_file"])))
+
+    # Here we add a wrapped module that implements the DFP Deployment
+    dfp_deployment_stage = pipeline.add_stage(
+        NonLinearModulesStage(config,
+                              module_config,
+                              input_port_name="input",
+                              output_port_name_prefix="output",
+                              output_port_count=output_port_count))
+
+    pipeline.add_edge(source_stage, dfp_deployment_stage)
+
+    if len(dfp_deployment_stage.output_ports) > 1:
+        tra_mntr_stage = MonitorStage(config, description="DFPTraining Pipeline rate", smoothing=0.001)
+        inf_mntr_stage = MonitorStage(config, description="DFPInference Pipeline rate", smoothing=0.001)
+
+        tra_mntr_stage = pipeline.add_stage(tra_mntr_stage)
+        inf_mntr_stage = pipeline.add_stage(inf_mntr_stage)
+
+        pipeline.add_edge(dfp_deployment_stage.output_ports[0], tra_mntr_stage)
+        pipeline.add_edge(dfp_deployment_stage.output_ports[1], inf_mntr_stage)
```

Comment:
Maybe I'm just not familiar enough with modules yet, but how do you know that when the module has 2 ports, port 0 will be training and port 1 will be inference?

I wonder if there a way to attach some properties to the module, or if there is a way to look this up in the config?